receiver/prometheusreceiver/internal/metrics.go

```diff
@@ -0,0 +1,51 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"context"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+var tagInstance, _ = tag.NewKey("instance")
+
+var statUpStatus = stats.Int64("up", "Whether the endpoint is alive or not", stats.UnitDimensionless)
```

Comment:
Should this be part of the metrics returned by the prometheus receiver itself rather than a self-observability metric?  I agree it could be implemented as either a self-obs metric or as an additional metric from the receiver, but there are a few advantages:

1. Users probably expect to see an 'up' metric when they make a pipeline with a prometheus receiver, and don't expect to have to do additional things.
2. I probably want to apply the same set of transformations to the 'up' metric that I want to apply to the rest of my prometheus metrics, since they should have the same resource labels (e.g. instance, job).
3. It will make it easier for us to pass prom compliance, since we won't need to route self-obs metrics to the PRW exporter.

Reply:
If this is a self-obs metric, will it still satisfy the prometheus compliance tests?